rules/validator.go

```diff
@@ -44,23 +47,30 @@ func NewValidator(opts ValidatorOptions) Validator {
 }
 
 func (v *validator) Validate(rs RuleSet) error {
-	if err := v.validate(rs); err != nil {
-		return v.wrapError(err)
-	}
-	return nil
-}
-
-func (v *validator) validate(rs RuleSet) error {
 	// Only the latest (a.k.a. the first) view needs to be validated
 	// because that is the view that may be invalid due to latest update.
 	latest, err := rs.Latest()
 	if err != nil {
-		return fmt.Errorf("could not get the latest ruleset snapshot: %v", err)
+		return v.wrapError(fmt.Errorf("could not get the latest ruleset snapshot: %v", err))
+	}
+	return v.ValidateSnapshot(latest)
+}
+
+func (v *validator) ValidateSnapshot(snapshot *RuleSetSnapshot) error {
+	if snapshot == nil {
```

Comment:
Seems odd to me that a nil snapshot is valid, will this occur in practice?

Reply:
No it shouldn't, this is just defensive programming though.